src/metpy/calc/basic.py

```diff
@@ -777,15 +778,180 @@ def smooth_gaussian(scalar_grid, n):
 
 
 @exporter.export
-@preprocess_xarray
-@units.wraps('=A', ('=A', None, None))
+@wrap_output_like(argument='scalar_grid', match_unit=True)
+def smooth_window(scalar_grid, window, passes=1, normalize_weights=True):
+    """Filter with an arbitrary window smoother.
+
+    Parameters
+    ----------
+    scalar_grid : array-like
+        N-dimensional scalar grid to be smoothed.
+
+    window : ndarray
+        The window to use in smoothing. Can have dimension less than or equal to N. If
+        dimension less than N, the scalar grid will be smoothed along its trailing dimensions.
+        Shape along each dimension must be odd.
+
+    passes : int
+        The number of times to apply the filter to the grid. Defaults to 1.
+
+    normalize_weights : bool
+        If true, divide the values in window by the sum of all values in the window to obtain
+        the normalized smoothing weights. If false, use supplied values directly as the
+        weights.
+
+    Returns
+    -------
+    array-like
+        The filtered scalar grid.
+
+    Notes
+    -----
+    This function can be applied multiple times to create a more smoothed field and will only
+    smooth the interior points, leaving the end points with their original values (this
+    function will leave an unsmoothed edge of size `(n - 1) / 2` for each `n` in the shape of
+    `window` around the data). If a masked value or NaN values exists in the array, it will
+    propagate to any point that uses that particular grid point in the smoothing calculation.
+    Applying the smoothing function multiple times will propogate NaNs further throughout the
+    domain.
+
+    See Also
+    --------
+    smooth_rectangular, smooth_circular, smooth_n_point, smooth_gaussian
+
+    """
+    def _pad(n):
+        # Return number of entries to pad given length along dimension.
+        return (n - 1) // 2
+
+    def _zero_to_none(x):
+        # Convert zero values to None, otherwise return what is given.
+        return x if x != 0 else None
+
+    def _offset(pad, k):
+        # Return padded slice offset by k entries
+        return slice(_zero_to_none(pad + k), _zero_to_none(-pad + k))
+
+    def _trailing_dims(indexer):
+        # Add ... to the front of an indexer, since we are working with trailing dimensions.
+        return (Ellipsis,) + tuple(indexer)
+
+    # Verify that shape in all dimensions is odd (need to have a neighboorhood around a
+    # central point)
+    if any((size % 2 == 0) for size in window.shape):
+        raise ValueError('The shape of the smoothing window must be odd in all dimensions.')
+
+    # Optionally normalize the supplied weighting window
+    if normalize_weights:
+        weights = window / np.sum(window)
+    else:
+        weights = window
+
+    # TODO: this is not lazy-loading/dask compatible, as it "densifies" the data
```

Comment:
Should that be make into an issue - afraid it'll get lost here.

Reply:
Do you think it merits its own issue, or just a note in https://github.com/Unidata/MetPy/issues/996?

Reply:
I think a note there is fine.